### PR TITLE
[query] implemented Graphite's `powSeries` function

### PR DESCRIPTION
**Pull request by DeanWahle**
**What this PR does / why we need it**:
This PR implements Graphite's `powSeries` function, which takes two or more series and pows their points. A constant line may be used.

**Special notes for your reviewer**:
[Graphite's powSeries function](https://graphite.readthedocs.io/en/stable/functions.html?highlight=powSeries#graphite.render.functions.powSeries) 
**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
no
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
no
```

### CODE REVIEW

File: src/query/graphite/native/aggregation_functions.go [hunk 1]
```diff
@@ -88,6 +88,12 @@ func minSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, er
 	return combineSeries(ctx, series, wrapPathExpr(minSeriesFnName, ts.SeriesList(series)), ts.Min)
 }
 
+// powSeries takes a list of series and returns a new series containing the
+// pow value across the series at each datapoint
+func powSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, error) {
```
Comment (robskillington) [inline src/query/graphite/native/aggregation_functions.go:93]: Should this be `func pow`?

File: src/query/graphite/native/functions.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ const (
 	minSeriesFnName      = "minSeries"
 	multiplyFnName       = "multiply"
 	multiplySeriesFnName = "multiplySeries"
+	powSeriesFnName      = "pow"
```
Comment (robskillington) [inline src/query/graphite/native/functions.go:64]: Should this perhaps be `"powSeries"`?

Comment (DeanWahle) [inline src/query/graphite/native/functions.go:64]:
```suggestion
	powSeriesFnName      = "powSeries"
```

Comment (DeanWahle) [reply src/query/graphite/native/functions.go]: @robskillington great catch, thanks.

**Review by robskillington — APPROVED** [review]
LGTM

**Comment by robskillington** [comment]
Looks like linter is failing FYI:
```

src/query/graphite/native/aggregation_functions.go:93:37: hugeParam: series is heavy (89 bytes); consider passing it by pointer (gocritic)
--
  | func powSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, error) {
  | ^
  | src/query/graphite/native/aggregation_functions_test.go:177: line is 102 characters (lll)
  | ctx       = common.NewContext(common.ContextOptions{Start: startTime, End: endTime, Engine: engine})

```

For `hugeParam` I would just avoid the linter complaining. i.e. `// nolint: hugeparam` the line above `func powSeries` and the inclusino of `multiplePathSpecs` struct.

File: src/query/graphite/native/aggregation_functions_test.go [hunk 3]
```diff
@@ -166,6 +166,36 @@ func TestStdDevSeries(t *testing.T) {
...
 
+func TestPowSeries(t *testing.T) {
+	var (
+		ctrl      = xgomock.NewController(t)
+		store     = storage.NewMockStorage(ctrl)
+		now       = time.Now().Truncate(time.Hour)
+		engine    = NewEngine(store)
+		startTime = now.Add(-3 * time.Minute)
+		endTime   = now.Add(-time.Minute)
+		ctx       = common.NewContext(common.ContextOptions{Start: startTime, End: endTime, Engine: engine})
```
Comment (DeanWahle) [inline src/query/graphite/native/aggregation_functions_test.go:177]:
```suggestion
		ctx       = common.NewContext(common.ContextOptions{
                                Start: startTime, 
                                End: endTime, 
                                Engine: engine,
                              })
```

File: src/query/graphite/native/aggregation_functions.go [hunk 4]
```diff
@@ -88,6 +88,12 @@ func minSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, er
 	return combineSeries(ctx, series, wrapPathExpr(minSeriesFnName, ts.SeriesList(series)), ts.Min)
 }
 
+// powSeries takes a list of series and returns a new series containing the
+// pow value across the series at each datapoint
```
Comment (DeanWahle) [inline src/query/graphite/native/aggregation_functions.go:92]:
```suggestion
// pow value across the series at each datapoint
// nolint: hugeParam
```

File: src/query/graphite/native/aggregation_functions.go [hunk 5]
```diff
@@ -88,6 +88,13 @@ func minSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, er
...
+// nolint: hugeParam
```
Comment (robskillington) [inline src/query/graphite/native/aggregation_functions.go:93]:
```suggestion
// nolint: gocritic
```

**Pull request merged**